### PR TITLE
Feature#81 display profile and logout button in a dropdown

### PR DESCRIPTION
**Pull request by MateuszBandura**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -153,6 +153,7 @@
           ".js",
           ".jsx",
           ".ts",
+          ".ts",
```
Comment (wps0) [inline .eslintrc:156]: ????
Isn't it already defined the line before?

File: src/components/Router.tsx [hunk 2]
```diff
@@ -27,8 +27,8 @@ function AddRouter() {
         <header className="App-header fixed w-full top-0 h-10 z-50">
           <Navigation />
         </header>
-        <div className="container w-full h-screen max-w-8xl mx-auto flex mt-12 z-10">
-          { isAdmin(userInfo) ? <SidebarAdmin /> : <Sidebar />}
+        <div className="container w-full h-full max-w-8xl mx-auto flex mt-12 z-10">
```
Comment (wps0) [inline src/components/Router.tsx:30]: I think min-h-screen would be a better choice

File: src/components/Sidebar.tsx [hunk 3]
```diff
@@ -33,10 +33,10 @@ function Sidebar() {
         </button>
       </div>
       <div
-        className={`lg:block z-30 inset-0 flex-none h-full w-8/12 sm:w-4/12 lg:h-auto bg-white ml-6 lg:mt-0 lg:ml-0 pb-10 lg:pt-0
-        lg:w-60 xl:w-72 shadow-sm dark:bg-gray-800 overflow-y-auto lg:overflow-y-visible ${toggle === true ? 'hidden' : 'fixed lg:static'}`}
+        className={`lg:block z-30 inset-0 flex-none h-full lg:h-auto bg-white ml-6 lg:mt-0 lg:ml-0 pb-10 lg:pt-0
+        w-72 shadow-sm dark:bg-gray-800 overflow-y-auto lg:overflow-y-visible ${toggle === true ? 'hidden' : 'fixed lg:static'}`}
```
Comment (wps0) [inline src/components/Sidebar.tsx:37]: ` ${toggle ? ...}` would be enough

File: src/components/SidebarAdmin.tsx [hunk 4]
```diff
@@ -17,10 +17,10 @@ function SidebarAdmin() {
         </button>
       </div>
       <div
-        className={`lg:block z-30 inset-0 flex-none h-full w-8/12 sm:w-4/12 lg:h-auto bg-white ml-6 lg:mt-0 lg:ml-0 pb-10 lg:pt-0
-        lg:w-60 xl:w-72 shadow-sm dark:bg-gray-800 overflow-y-auto lg:overflow-y-visible ${toggle === true ? 'hidden' : 'fixed lg:static'}`}
+        className={`lg:block z-30 inset-0 flex-none h-full lg:h-auto bg-white ml-6 lg:mt-0 lg:ml-0 pb-10 lg:pt-0
+        w-72 shadow-sm dark:bg-gray-800 overflow-y-auto lg:overflow-y-visible ${toggle === true ? 'hidden' : 'fixed lg:static'}`}
```
Comment (wps0) [inline src/components/SidebarAdmin.tsx:21]: ` ${toggle ? ...}` would be enough

File: src/components/UserDropdown.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import React, { FC, useState } from 'react';
+import { Link } from 'react-router-dom';
+
+interface UserDropdownProps {
+  title: string | undefined;
+  logOut: () => void;
+}
+
+const UserDropdown: FC<UserDropdownProps> = ({ title, logOut }) => {
+  const [Open, setOpen] = useState(false);
```
Comment (wps0) [inline src/components/UserDropdown.tsx:10]: It'd be better if you had used lower-case here.

File: src/components/UserDropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        onClick={() => setOpen(!Open)}
+      >
+        {title}
+      </div>
+      {Open && (
+        <div
+          className="dropdown block md:absolute w-auto overflow-hidden md:-translate-y-0
+        bg-gray-600 text-white border border-gray-800 rounded-sm"
+        >
+          <div className="menu-item border-b border-gray-700">
```
Comment (wps0) [inline src/components/UserDropdown.tsx:26]: Try to extract the parts responsible for single dropdown option to another component to make the dropdown reusable. I'd also suggest the dropdown component accept list of dropdown options and urls to which they redirect. Right now, you can't use the component anywhere else than in navbar.

File: src/index.css [hunk 7]
```diff
@@ -1,3 +1,26 @@
...
+  .dropdown {
+    top: 50px;
+    transform: translateX(15%);
+  }
+}
+
+@media only screen and (max-width: 767px) {
+  .dropdown {
+    top: 50px;
+    /* transform: translateY(70%); */
```
Comment (wps0) [inline src/index.css:15]: If it isn't used, it can be deleted.

File: src/index.css [hunk 8]
```diff
@@ -2,9 +2,32 @@
...
+}
+
 @layer components {
   .btn-nav {
-    @apply my-1 py-1 md:py-0 pl-4 text-gray-200 transition duration-500 ease-in-out hover:text-indigo-500 md:mr-4 md:my-0 border-b border-t md:border-t-0 md:border-b-0 md:border-l border-gray-400 cursor-pointer;
+    @apply px-4 py-2 md:py-0 text-gray-200 transition duration-500 ease-in-out hover:text-indigo-500 border-t md:border-t-0 md:border-l border-gray-400 cursor-pointer;
+  }
+
+  .btn-nav-end {
+    @apply px-4 py-2 md:py-0 text-gray-200 transition duration-500 ease-in-out hover:text-indigo-500 border-t border-b md:border-t-0 md:border-b-0 md:border-l md:border-r border-gray-400 cursor-pointer;
```
Comment (DLowHP) [inline src/index.css:30]: Could be .btn-nav:last-child

**Pull request merged**